test/functional/rpc_signrawtransaction.py

```diff
@@ -168,6 +169,44 @@ def witness_script_test(self):
         assert 'complete' in spending_tx_signed
         assert_equal(spending_tx_signed['complete'], True)
 
+        # Now try with a P2PKH script as the witnessScript
+        embedded_addr_info = self.nodes[1].getaddressinfo(self.nodes[1].getnewaddress('', 'legacy'))
+        embedded_privkey = self.nodes[1].dumpprivkey(embedded_addr_info['address'])
+        witness_script = embedded_addr_info['scriptPubKey']
+        redeem_script = CScript([OP_0, sha256(check_script(witness_script))]).hex()
+        addr = script_to_p2sh(redeem_script)
+        script_pub_key = self.nodes[1].validateaddress(addr)['scriptPubKey']
+        # Fund that address
+        txid = self.nodes[0].sendtoaddress(addr, 10)
+        vout = find_vout_for_address(self.nodes[0], txid, addr)
+        self.nodes[0].generate(1)
+        # Now create and sign a transaction spending that output on node[0], which doesn't know the scripts or keys
+        spending_tx = self.nodes[0].createrawtransaction([{'txid': txid, 'vout': vout}], {self.nodes[1].getnewaddress(): Decimal("9.999")})
+        spending_tx_signed = self.nodes[0].signrawtransactionwithkey(spending_tx, [embedded_privkey], [{'txid': txid, 'vout': vout, 'scriptPubKey': script_pub_key, 'redeemScript': redeem_script, 'witnessScript': witness_script, 'amount': 10}])
```

Comment:
verified that reverting the change in rawtransaction_util.cpp::219 causes the test to fail here with `redeemScript does not correspond to witnessScript (-8)`